### ISSUE TITLE
java.lang.IllegalStateException: sequence overflow agentId:asp-batch-01

### ISSUE THREAD

**Issue by julienBaleizao** [naver/pinpoint#6419]
**What version of pinpoint are you using?**
1.8.5

**Describe the bug?**
When I want to access the transaction list for selected request I get this error:
```
2020-01-14 17:12:51 [WARN ](c.n.p.c.s.b.s.t.v.SpanDecoderV0    ) sequence overflow. firstSpanEventSequence:-20532 basicSpan:SpanBo{version=0, agentId='asp-batch-01', applicationId='asp-batch', agentStartTime=1579017448785, transactionId=TransactionId{agentId='asp-batch-01', agentStartTime=1579017448785, transactionSequence=5}, spanId=-4529154742101988419, parentSpanId=0, parentApplicationId='null', parentApplicationServiceType=0, startTime=0, elapsed=0, rpc='null', serviceType=0, endPoint='null', apiId=0, annotationBoList=[], flag=0, errCode=0, spanEventBoList=[], collectorAcceptTime=1579018083147, hasException=false, exceptionId=0, exceptionMessage='null', exceptionClass='null', applicationServiceType=null, acceptorHost='null', remoteAddr='null', loggingTransactionInfo=0}
2020-01-14 17:12:51 [WARN ](.n.p.w.c.ControllerExceptionHandler) Failed to execute controller methods. message:sequence overflow agentId:asp-batch-01, request:{method=POST, heads={content-length=[57], referer=[http://10.90.248.53:8082/], accept-language=[fr,fr-FR;q=0.8,en-US;q=0.5,en;q=0.3], cookie=[JSESSIONID.eac7b6d5=node018fabrrgw77yxfenpce7hbznz0.node0; JSESSIONID=10inkmxhyoix4vcr3yfhu9w0c; _ga=GA1.1.1165543182.1578928913; _gid=GA1.1.184491134.1578928913; _gat=1], origin=[http://10.90.248.53:8082], x-forwarded-for=[10.112.245.100], accept=[application/json, text/plain, */*], via=[1.1 ypsprx01s.priv.atos.fr (squid/3.1.23)], host=[10.90.248.53:8082], content-type=[application/x-www-form-urlencoded], connection=[keep-alive], cache-control=[max-age=259200], accept-encoding=[gzip, deflate], user-agent=[Mozilla/5.0 (Windows NT 6.1; Win64; x64; rv:71.0) Gecko/20100101 Firefox/71.0]}, parameters={I0=[asp-batch-01^1579017448785^5], T0=[1579018083147], R0=[29563]}, url=http://10.90.248.53:8082/transactionmetadata.pinpoint}.
java.lang.IllegalStateException: sequence overflow agentId:asp-batch-01
        at com.navercorp.pinpoint.common.server.bo.serializer.trace.v2.SpanDecoderV0.readQualifier(SpanDecoderV0.java:405)
        at com.navercorp.pinpoint.common.server.bo.serializer.trace.v2.SpanDecoderV0.readSpan(SpanDecoderV0.java:74)
        at com.navercorp.pinpoint.common.server.bo.serializer.trace.v2.SpanDecoderV0.decode(SpanDecoderV0.java:37)
        at com.navercorp.pinpoint.web.mapper.SpanMapperV2.mapRow(SpanMapperV2.java:101)
        at com.navercorp.pinpoint.web.mapper.SpanMapperV2.mapRow(SpanMapperV2.java:54)
        at com.navercorp.pinpoint.common.hbase.HbaseTemplate2$5.doInTable(HbaseTemplate2.java:305)
        at com.navercorp.pinpoint.common.hbase.HbaseTemplate2$5.doInTable(HbaseTemplate2.java:299)
        at com.navercorp.pinpoint.common.hbase.HbaseTemplate2.execute(HbaseTemplate2.java:769)
        at com.navercorp.pinpoint.common.hbase.HbaseTemplate2.get(HbaseTemplate2.java:299)
        at com.navercorp.pinpoint.web.dao.hbase.HbaseTraceDaoV2.select0(HbaseTraceDaoV2.java:148)
        at com.navercorp.pinpoint.web.dao.hbase.HbaseTraceDaoV2.partitionSelect(HbaseTraceDaoV2.java:130)
        at com.navercorp.pinpoint.web.dao.hbase.HbaseTraceDaoV2.selectSpans(HbaseTraceDaoV2.java:97)
        at com.navercorp.pinpoint.web.dao.hbase.HbaseTraceDaoV2.selectSpans(HbaseTraceDaoV2.java:87)
        at com.navercorp.pinpoint.web.service.ScatterChartServiceImpl.selectTransactionMetadata(ScatterChartServiceImpl.java:107)
        at com.navercorp.pinpoint.web.controller.ScatterChartController.transactionmetadata(ScatterChartController.java:103)
        at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
        at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
        at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.lang.reflect.Method.invoke(Method.java:498)
        at org.springframework.web.method.support.InvocableHandlerMethod.doInvoke(InvocableHandlerMethod.java:205)
        at org.springframework.web.method.support.InvocableHandlerMethod.invokeForRequest(InvocableHandlerMethod.java:133)
        at org.springframework.web.servlet.mvc.method.annotation.ServletInvocableHandlerMethod.invokeAndHandle(ServletInvocableHandlerMethod.java:97)
        at org.springframework.web.servlet.mvc.method.annotation.RequestMappingHandlerAdapter.invokeHandlerMethod(RequestMappingHandlerAdapter.java:849)
        at org.springframework.web.servlet.mvc.method.annotation.RequestMappingHandlerAdapter.handleInternal(RequestMappingHandlerAdapter.java:760)
        at org.springframework.web.servlet.mvc.method.AbstractHandlerMethodAdapter.handle(AbstractHandlerMethodAdapter.java:85)
        at org.springframework.web.servlet.DispatcherServlet.doDispatch(DispatcherServlet.java:967)
        at org.springframework.web.servlet.DispatcherServlet.doService(DispatcherServlet.java:901)
        at org.springframework.web.servlet.FrameworkServlet.processRequest(FrameworkServlet.java:970)
        at org.springframework.web.servlet.FrameworkServlet.doPost(FrameworkServlet.java:872)
        at javax.servlet.http.HttpServlet.service(HttpServlet.java:660)
        at org.springframework.web.servlet.FrameworkServlet.service(FrameworkServlet.java:846)
        at javax.servlet.http.HttpServlet.service(HttpServlet.java:741)
        at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:231)
        at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:166)
        at org.apache.tomcat.websocket.server.WsFilter.doFilter(WsFilter.java:53)
        at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:193)
        at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:166)
        at com.navercorp.pinpoint.web.servlet.NoCacheFilter.doFilter(NoCacheFilter.java:50)
        at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:193)
        at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:166)
        at org.springframework.web.filter.CharacterEncodingFilter.doFilterInternal(CharacterEncodingFilter.java:197)
        at org.springframework.web.filter.OncePerRequestFilter.doFilter(OncePerRequestFilter.java:107)
        at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:193)
        at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:166)
        at org.apache.catalina.core.StandardWrapperValve.invoke(StandardWrapperValve.java:202)
        at org.apache.catalina.core.StandardContextValve.invoke(StandardContextValve.java:96)
        at org.apache.catalina.authenticator.AuthenticatorBase.invoke(AuthenticatorBase.java:541)
        at org.apache.catalina.core.StandardHostValve.invoke(StandardHostValve.java:139)
        at org.apache.catalina.valves.ErrorReportValve.invoke(ErrorReportValve.java:92)
        at org.apache.catalina.valves.AbstractAccessLogValve.invoke(AbstractAccessLogValve.java:678)
        at org.apache.catalina.core.StandardEngineValve.invoke(StandardEngineValve.java:74)
        at org.apache.catalina.connector.CoyoteAdapter.service(CoyoteAdapter.java:343)
        at org.apache.coyote.http11.Http11Processor.service(Http11Processor.java:367)
        at org.apache.coyote.AbstractProcessorLight.process(AbstractProcessorLight.java:65)
        at org.apache.coyote.AbstractProtocol$ConnectionHandler.process(AbstractProtocol.java:860)
        at org.apache.tomcat.util.net.NioEndpoint$SocketProcessor.doRun(NioEndpoint.java:1598)
        at org.apache.tomcat.util.net.SocketProcessorBase.run(SocketProcessorBase.java:49)
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
        at org.apache.tomcat.util.threads.TaskThread$WrappingRunnable.run(TaskThread.java:61)
        at java.lang.Thread.run(Thread.java:748)
```
**What did you do to trigger the bug?**
I have selected some request on the graphic on the main page.

**Expected behavior**
I get the list of the transaction for the selected request.

**Screenshots**
![pinpoint](https://user-images.githubusercontent.com/25363471/72363287-ddd70d80-36f4-11ea-8be2-93a462c70e96.PNG)

**Comment by koo-taejin** [comment]
hi @julienBaleizao 

This issue expects to have resolved in 2.0.0 version.
So I recommend you to use 2.0.0-RC2.
(You must upgrade agent, collector and web)

In addition, the root cause of the problem is having too much traces. (Approximately over 30,000 traces.)
In this case, It is hard to find problem, even if you watch callstack. 
If you disable plug-ins that do not judge important, you will be able to find meaningful information quickly.

thanks :)

**Comment by emeroad** [comment]
Resolved at #6324

**Comment by julienBaleizao** [comment]
Hi,
Following your answer I have installed pinpoint 2.0.0-RC2 but now I have this error:
`2020-01-15 14:09:59 [WARN ](c.n.p.c.h.g.GrpcSpanChunkHandler   ) Failed to handle spanChunk=version: 1 transactionId { agentStartTime: 1579093611493 sequence: 1 } spanId: -6993923593182931627 endPoint: "localhost:8002" spanEvent { sequence: -12750 depth: 5 endElapsed: 1 serviceType: 5071 apiId: 148 } spanEvent { sequence: -12747 startElapsed: 1 endElapsed: 1 serviceType: 5071 apiId: 148 } spanEvent { sequence: -12746 depth: 6 serviceType: 2500 annotation { key: 20 value { intStringStringValue { intValue: 18 } } } apiId: 669 nextEvent { messageEvent { endPoint: "localhost:5432" destinationId: "ch-energie" } } } spanEvent { sequence: -12745 endElapsed: 1 serviceType: 2501 annotation { key: 20 value { intStringStringValue { intValue: 18 stringValue2 { value: "1833576981" } } } } apiId: 681 nextEvent { messageEvent { endPoint: "localhost:5432" destinationId: "ch-energie" } } } spanEvent { sequence: -12744 depth: 5 startElapsed: 1 endElapsed: 1 serviceType: 5071 apiId: 148 } spanEvent { sequence: -12743 depth: 6 serviceType: 2500 annotation { key: 20 value { intStringStringValue { intValue: 18 } } } apiId: 669 nextEvent { messageEvent { endPoint: "localhost:5432" destinationId: "ch-energie" } } } spanEvent { sequence: -12742 endElapsed: 1 serviceType: 2501 annotation { key: 20 value { intStringStringValue { intValue: 18 stringValue2 { value: "1868235094" } } } } apiId: 681 nextEvent { messageEvent { endPoint: "localhost:5432" destinationId: "ch-energie" } } } spanEvent { sequence: -12741 depth: 5 startElapsed: 1 endElapsed: 1 serviceType: 5071 apiId: 148 } spanEvent { sequence: -12740 depth: 6 serviceType: 2500 annotation { key: 20 value { intStringStringValue { intValue: 18 } } } apiId: 669 nextEvent { messageEvent { endPoint: "localhost:5432" destinationId: "ch-energie" } } } spanEvent { sequence: -12739 endElapsed: 1 serviceType: 2501 annotation { key: 20 value { intStringStringValue { intValue: 18 stringValue2 { value: "1838789228" } } } } apiId: 681 nextEvent { messageEvent { endPoint: "localhost:5432" destinationId: "ch-energie" } } } spanEvent { sequence: -12738 depth: 5 startElapsed: 1 endElapsed: 1 serviceType: 5071 apiId: 148 } spanEvent { sequence: -12737 depth: 6 serviceType: 2500 annotation { key: 20 value { intStringStringValue { intValue: 18 } } } apiId: 669 nextEvent { messageEvent { endPoint: "localhost:5432" destinationId: "ch-energie" } } } spanEvent { sequence: -12736 endElapsed: 1 serviceType: 2501 annotation { key: 20 value { intStringStringValue { intValue: 18 stringValue2 { value: "1821871669" } } } } apiId: 681 nextEvent { messageEvent { endPoint: "localhost:5432" destinationId: "ch-energie" } } } spanEvent { sequence: -12735 depth: 5 startElapsed: 1 endElapsed: 1 serviceType: 5071 apiId: 148 } spanEvent { sequence: -12734 depth: 6 serviceType: 2500 annotation { key: 20 value { intStringStringValue { intValue: 18 } } } apiId: 669 nextEvent { messageEvent { endPoint: "localhost:5432" destinationId: "ch-energie" } } } spanEvent { sequence: -12733 serviceType: 2501 annotation { key: 20 value { intStringStringValue { intValue: 18 stringValue2 { value: "1853864693" } } } } apiId: 681 nextEvent { messageEvent { endPoint: "localhost:5432" destinationId: "ch-energie" } } } spanEvent { sequence: -12732 depth: 5 startElapsed: 1 serviceType: 5071 apiId: 148 } spanEvent { sequence: -12731 depth: 6 serviceType: 2500 annotation { key: 20 value { intStringStringValue { intValue: 18 } } } apiId: 669 nextEvent { messageEvent { endPoint: "localhost:5432" destinationId: "ch-energie" } } } spanEvent { sequence: -12730 serviceType: 2501 annotation { key: 20 value { intStringStringValue { intValue: 18 stringValue2 { value: "1820428566" } } } } apiId: 681 nextEvent { messageEvent { endPoint: "localhost:5432" destinationId: "ch-energie" } } } spanEvent { sequence: -12728 startElapsed: 1 serviceType: 2500 annotation { key: 20 value { intStringStringValue { intValue: 18 } } } apiId: 669 nextEvent { messageEvent { endPoint: "localhost:5432" destinationId: "ch-energie" } } } applicationServiceType: 1010 keyTime: 1579093800266
java.lang.NullPointerException
`
Do you know how I can solve it ?
